yellowstone-grpc-geyser/src/monitor.rs

```diff
@@ -0,0 +1,52 @@
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
+
+use once_cell::sync::Lazy;
+use solana_client::nonblocking::rpc_client::RpcClient;
+use solana_rpc_client_api::{client_error, request};
+use tokio::time::interval;
+
+pub const HEALTH_CHECK_SLOT_DISTANCE: u64 = 100;
+pub const IS_NODE_UNHEALTHY: Lazy<Arc<AtomicBool>> = Lazy::new(|| Arc::new(AtomicBool::new(false)));
```

Comment:
I considered disconnecting clients only when the node was unhealthy by some amount of time. But I think that was unnecessarily complicated. If a node is behind by 100 slots, then it must have been unhealthy for about 30 seconds, which is enough.